src/main/java/mate/academy/library/controllers/CategoryController.java

```diff
@@ -0,0 +1,76 @@
+package mate.academy.library.controllers;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import mate.academy.library.dto.book.BookDtoWithoutCategoryIds;
+import mate.academy.library.dto.category.CategoryDto;
+import mate.academy.library.service.BookService;
+import mate.academy.library.service.CategoryService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import java.util.List;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping(value = "/categories")
+@Tag(name = "Categories management", description = "Endpoint for mapping categories")
+public class CategoryController {
+    private final CategoryService categoryService;
+    private final BookService bookService;
+
+    @PostMapping
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @Operation(summary = "Create new category", description = "Creates a new category entry in the database")
+    public CategoryDto createCategory(@RequestBody CategoryDto categoryDto) {
+        return categoryService.save(categoryDto);
+    }
+
+    @GetMapping
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @Operation(summary = "Get all categories", description = "Retrieves all categories from the database")
+    public List<CategoryDto> getAll(Pageable pageable) {
+        return categoryService.findAll(pageable);
+    }
+
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping("/{id}")
+    @Operation(summary = "Get category by id", description = "Retrieves a category by its identifier")
+    public CategoryDto getCategoryById(@PathVariable Long id) {

```

Comment:
```suggestion
    public CategoryDto getCategoryById(@PathVariable @Positive Long id) {

```
Don't forget to add `@Validated` on the class